dashboard/modules/tune/tune_head.py

```diff
@@ -0,0 +1,233 @@
+import logging
+import datetime
+import copy
+import os
+import aiohttp.web
+
+import ray.new_dashboard.modules.tune.tune_consts \
+    as tune_consts
+import ray.new_dashboard.utils as dashboard_utils
+from ray.new_dashboard.utils import async_loop_forever, rest_response
+
+try:
+    from ray.tune import Analysis
+    from tensorboard import program
+except ImportError:
+    Analysis = None
+
+logger = logging.getLogger(__name__)
+routes = dashboard_utils.ClassMethodRouteTable
+
+
+class TuneController(dashboard_utils.DashboardHeadModule):
+    def __init__(self, dashboard_head):
+        """
+        This dashboard module is responsible for enabling the Tune tab of
+        the dashboard. To do so, it periodically scrapes Tune output logs,
+        transforms them, and serves them up over an API.
+
+        Args
+            logdir (str): Directory path to save the status information of
+                            jobs and trials.
+        """
+        super().__init__(dashboard_head)
+        self._logdir = None
+        self._trial_records = {}
+        self._trials_available = False
+        self._tensor_board_dir = ""
+        self._enable_tensor_board = False
+        self._errors = {}
+
+    @routes.get("/tune/info")
+    async def tune_info(self, req) -> aiohttp.web.Response:
+        stats = self.get_stats()
+        return await rest_response(
+            success=True, message="Fetched tune info", result=stats)
+
+    @routes.get("/tune/availability")
+    async def get_availability(self, req) -> aiohttp.web.Response:
+        availability = self._get_availability()
+        return await rest_response(
+            success=True,
+            message="Fetched tune availability",
+            result=availability)
+
+    @routes.get("/tune/set_experiment")
+    async def set_tune_experiment(self, req) -> aiohttp.web.Response:
+        try:
+            experiment = req.query["experiment"]
+        except KeyError:
+            return await rest_response(success=False, message="Bad request")
```

Comment:
The `@routes.get` decorator catches the handler exception and returns `rest_response(success=False, message=<exception stack>)`. This `try ... except ...` do the same thing, but makes the response much cleaner `rest_response(success=False, message="Bad request")`.